File: src/main/java/com/intellij/tapestry/core/util/chain/Command.java
package com.intellij.tapestry.core.util.chain;

/**
 * A unit of work in a {@link Chain}.
 */
public interface Command {

    /**
     * Executes this command.
     *
     * @param context the shared execution context.
     * @return {@code true} if processing is complete and the chain should stop, {@code false} to continue.
     * @throws Exception if an error occurs.
     */
    boolean execute(Context context) throws Exception;
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/core/util/chain/Command.kt
package com.github.rar91279.plugin.tapestry.core.util.chain

/**
 * A unit of work in a [Chain].
 */
interface Command {

    /**
     * Executes this command.
     *
     * @param context the shared execution context.
     * @return `true` if processing is complete and the chain should stop, `false` to continue.
     */
    @Throws(Exception::class)
    fun execute(context: Context): Boolean
}
